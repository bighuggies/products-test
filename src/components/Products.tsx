import { observer } from 'mobx-react';
import * as React from 'react';

import { IProduct } from '../models/Product';

import Product from './Product';

interface IProps {
  products: IProduct[];
  filter: string;
  onFilter: (filter: string) => void;
}

class Products extends React.Component<IProps> {
  public updateFilter = (e: React.FormEvent<HTMLInputElement>) => {
    this.props.onFilter(e.currentTarget.value);
  };

  public render() {
    return (
      <div>
        <input
          type="search"
          value={this.props.filter}
          onInput={this.updateFilter}
        />

        <ul>
          {this.props.products.map(p => (
            <li key={p.id}>
              <Product product={p} />
            </li>
          ))}
        </ul>
      </div>
    );
  }
}

export default observer(Products);
