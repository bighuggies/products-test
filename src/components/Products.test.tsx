import { mount } from 'enzyme';
import * as React from 'react';

import Products from './Products';

// tslint:disable no-empty
const noop = () => {};

test('renders without crashing', () => {
  const mounted = mount(
    <Products
      products={[{ id: '1', description: '1', title: '1', categories: [] }]}
      filter={''}
      onFilter={noop}
    />
  );

  expect(mounted.debug()).toMatchSnapshot();
});

test('renders correctly without products', () => {
  const mounted = mount(<Products products={[]} filter={''} onFilter={noop} />);

  expect(mounted.debug()).toMatchSnapshot();
});
