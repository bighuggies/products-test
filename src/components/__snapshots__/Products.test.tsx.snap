// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly without products 1`] = `
"<Products products={{...}} filter=\\"\\" onFilter={[Function: noop]}>
  <styled.div>
    <div className=\\"sc-ifAKCX hoKjwN\\">
      <styled.input type=\\"search\\" value=\\"\\" onInput={[Function]}>
        <input className=\\"sc-bdVaJa glpHrW\\" type=\\"search\\" value=\\"\\" onInput={[Function]} />
      </styled.input>
      <p>
        No products in this category
      </p>
    </div>
  </styled.div>
</Products>"
`;

exports[`renders without crashing 1`] = `
"<Products products={{...}} filter=\\"\\" onFilter={[Function: noop]}>
  <styled.div>
    <div className=\\"sc-ifAKCX hoKjwN\\">
      <styled.input type=\\"search\\" value=\\"\\" onInput={[Function]}>
        <input className=\\"sc-bdVaJa glpHrW\\" type=\\"search\\" value=\\"\\" onInput={[Function]} />
      </styled.input>
      <styled.ul>
        <ul className=\\"sc-bxivhb ktajjB\\">
          <li>
            <Product product={{...}}>
              <div>
                <styled.button className=\\"\\" onClick={[Function]} aria-expanded={false} aria-controls=\\"1-description\\">
                  <button className=\\"sc-bwzfXH fJXyvc\\" onClick={[Function]} aria-expanded={false} aria-controls=\\"1-description\\">
                    1
                  </button>
                </styled.button>
              </div>
            </Product>
          </li>
        </ul>
      </styled.ul>
    </div>
  </styled.div>
</Products>"
`;
